### PR TITLE
DAOS-4873 server: client stack on server

### PR DESCRIPTION
**Pull request by gnailzenh**
DAOS server stack relies on client stack for many reasons:
- rebuild, reintegration, rebalance
- object enumeration
- EC aggregation

This patch moves those functions in src/iosrv/srv_cli.c to the
components they belonging to:
- src/object/srv_cli.c
- src/container/srv_cli.c
- src/pool/srv_cli.c

It also includes a few cleanups:
- prefix of client functions used by server (only) are changed
  from "dc_" to "dsc_"
- dc_cont_local_open/close are moved to src/container/srv_cli.c
  because they are only used by server stack
- dc_pool_local_open/close are moved to src/pool/srv_cli.c
  because they are only used by server stack
- Functions exported to other modules should have dc/ds/dsc_ as
  prefix, a few container IV functions are renamed due to this.
- Task body function of DAOS API are using _task as postfix to
  avoid naming conflicts (only renamed a few, the rest will be
  renamed in a follow-on patch)

TODO:
- add _task postfix for all task body functions of DAOS API

Signed-off-by: Liang Zhen <liang.zhen@intel.com>

### CODE REVIEW

File: src/container/container_iv.c [hunk 1]
```diff
@@ -1092,6 +1084,40 @@ cont_iv_prop_fetch(struct ds_iv_ns *ns, uuid_t cont_hdl_uuid,
...
+{
+	D_ASSERT(dss_get_module_info()->dmi_xs_id == 0);
+	return cont_iv_snapshots_fetch(ns, cont_uuid, snapshots, snap_count);
+}
+
+int
+ds_cont_revoke_snaps(struct ds_iv_ns *ns, uuid_t cont_uuid,
+		     unsigned int shortcut, unsigned int sync_mode)
+{
+	D_ASSERT(dss_get_module_info()->dmi_xs_id == 0);
```
Comment (wangdi1) [inline src/container/container_iv.c:1102]: probably these assert(xs_id == 0) should be moved to iv_op_internal(), i.e. only stream 0 is allowed to do iv ops.

File: src/container/container_iv.c [hunk 2]
```diff
@@ -1092,6 +1084,40 @@ cont_iv_prop_fetch(struct ds_iv_ns *ns, uuid_t cont_hdl_uuid,
 	return rc;
 }
 
+/*
+ * exported APIs
+ */
+int
```
Comment (NiuYawei) [inline src/container/container_iv.c:1090]: Why not just rename the 'cont_iv_snapshots_fetch' to 'ds_cont_fetch_snaps'?

File: src/container/container_iv.c [hunk 3]
```diff
@@ -1092,6 +1084,40 @@ cont_iv_prop_fetch(struct ds_iv_ns *ns, uuid_t cont_hdl_uuid,
...
+ds_cont_fetch_snaps(struct ds_iv_ns *ns, uuid_t cont_uuid,
+		    uint64_t **snapshots, int *snap_count)
+{
+	D_ASSERT(dss_get_module_info()->dmi_xs_id == 0);
+	return cont_iv_snapshots_fetch(ns, cont_uuid, snapshots, snap_count);
+}
+
+int
+ds_cont_revoke_snaps(struct ds_iv_ns *ns, uuid_t cont_uuid,
```
Comment (NiuYawei) [inline src/container/container_iv.c:1099]: Ditto. Rename the original function name looks better to me.

File: src/container/container_iv.c [hunk 4]
```diff
@@ -1092,6 +1084,40 @@ cont_iv_prop_fetch(struct ds_iv_ns *ns, uuid_t cont_hdl_uuid,
...
+		     unsigned int shortcut, unsigned int sync_mode)
+{
+	D_ASSERT(dss_get_module_info()->dmi_xs_id == 0);
+	return cont_iv_snapshot_invalidate(ns, cont_uuid, shortcut, sync_mode);
+}
+
+int
+ds_cont_fetch_prop(struct ds_iv_ns *ns, uuid_t coh_uuid,
```
Comment (NiuYawei) [inline src/container/container_iv.c:1107]: Ditto.

File: src/container/container_iv.c [hunk 5]
```diff
@@ -1092,6 +1084,40 @@ cont_iv_prop_fetch(struct ds_iv_ns *ns, uuid_t cont_hdl_uuid,
...
+		   daos_prop_t *cont_prop)
+{
+	/* NB: it can be called from any xstream */
+	return cont_iv_prop_fetch(ns, coh_uuid, cont_prop);
+}
+
+int
+ds_cont_find_hdl(uuid_t po_uuid, uuid_t coh_uuid, struct ds_cont_hdl **coh_p)
```
Comment (NiuYawei) [inline src/container/container_iv.c:1115]: Ditto.

File: src/container/container_iv.c (hunk 2 above)
Comment (gnailzenh) [reply src/container/container_iv.c:1090]: yes, I just wanted to minimize the change, but I agree we don't have a lot of callers for these functions at all.

**Pull request merged**